Add more settings.

gassmoeller opened the pull request:
More specifically add the options to hide default values (to clean up the windows, and only see important options), and to change the window font (e.g. high DPI screens). Both options are available via a settings dialog, and the new toolbar that also allows opening and saving files. I have found the toolbar to be convenient, because it speeds up using the program.

gassmoeller committed:
Add preliminary settings dialog

Implement change font option

Preliminary hiding feature

Add toolbar

bangerth commented on mainwindow.cpp:
```diff
@@ -270,6 +276,86 @@ namespace dealii
+
+      if (hide_default_values)
+        {
+        for (int i = 0; i < tree_widget->topLevelItemCount(); ++i)
```
are you running the indentation script for the GUI as well?

bangerth replied:
If not, then ignore the indentation here and use a follow-up patch to just indent the whole thing.

bangerth commented on mainwindow.cpp:
```diff
@@ -270,6 +276,86 @@ namespace dealii
+
+
+
+    bool MainWindow::hide_default_item(QTreeWidgetItem *item)
```
I think the name is misleading. You are hiding items whose values equal the default. Maybe `hide_items_with_default_value`()?

bangerth commented on mainwindow.cpp:
```diff
@@ -270,6 +276,86 @@ namespace dealii
+        }
+      else
+        {
+          for (int i = 0; i < item->childCount(); ++i)
```
maybe add a comment that this recurses into subsections

bangerth reviewed (commented):
I don't really have a whole lot of useful content to contribute, knowing little about Qt.

One question I have is whether it is more useful to have the "Hide defaulted parameters" setting in a dialog box, or whether it would be easier to just have a global button somewhere that is always visible?

gassmoeller committed:
Address comments

gassmoeller commented:
I addressed the comments. 
Currently, I have created both a 'Settings' Menu option that opens a dialog, and a toolbar that is always visible and contains the new options. The 'Settings' menu will be most useful if there will be more options in the longer term, currently the toolbar is faster to use (just one click).

I did not use the indent script (only my eclipse auto indent), because there are many places that do not follow our convention, also in regard to the placement of comments. I will open a new PR that fixes indentation and comments for the whole project.
